ipn/ipnlocal/cert.go

```diff
@@ -146,46 +146,56 @@ var errCertExpired = errors.New("cert expired")
 
 func (b *LocalBackend) getCertStore(dir string) certStore {
 	if hostinfo.GetEnvType() == hostinfo.Kubernetes && dir == "/tmp" {
-		return certStateStore{b.store}
+		return certStateStore{StateStore: b.store}
 	}
-	return certFileStore(dir)
+	return certFileStore{dir: dir}
 }
 
 // certFileStore implements certStore by storing the cert & key files in the named directory.
-type certFileStore string // dir
+type certFileStore struct {
+	dir string
+
+	// This field allows a test to inject additional CA root(s) for certificate
+	// verification. If nil or empty, the default system pool is used.
```

Comment:
// testRoots, if non-empty for tests, is the PEM-formatted blah blah.

(is it PEM? I don't know!)

Reply:
It is. Updated.